x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/dataframe/DataFrameAnalyticsDest.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.core.ml.dataframe;
+
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.xcontent.ConstructingObjectParser;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.xpack.core.ml.utils.ExceptionsHelper;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public class DataFrameAnalyticsDest implements Writeable, ToXContentObject {
+
+    public static final ParseField INDEX = new ParseField("index");
+
+    public static final ConstructingObjectParser<DataFrameAnalyticsDest, Void> STRICT_PARSER = createParser(false);
+    public static final ConstructingObjectParser<DataFrameAnalyticsDest, Void> LENIENT_PARSER = createParser(true);
```

Comment:
are they used? I saw you call `createParser` directly.

(nit: I prefer implementing fromXContent)